core/matchers/function-spy-matcher.ts

```diff
@@ -19,122 +19,92 @@ export class FunctionSpyMatcher {
-        this._spy,
-        true,
-        expectedCallCount,
-        SpyCallCountType.Exactly
-      );
-    }
+    this._throwCallCountError(
+      count => count !== expectedCallCount,
+      expectedCallCount,
+      SpyCallCountType.Exactly,
+      true
+    );
 
     return new FunctionSpyCallCountMatcher();
   }
 
   public anythingBut(unexpectedCallCount: number): FunctionSpyCallCountMatcher {
-    if (unexpectedCallCount < 1) {
-      throw new TypeError("unexpectedCallCount must be greater than 0.");
-    }
+    this._validateCallCount(unexpectedCallCount, "unexpectedCallCount");
 
-    if (
-      this._expectedArguments &&
-      this._getArgumentsMatching(this._expectedArguments).length ===
-        unexpectedCallCount
-    ) {
-      throw new FunctionCallCountMatchError(
-        this._spy,
-        false,
-        unexpectedCallCount,
-        SpyCallCountType.Exactly,
-        this._expectedArguments
-      );
-    } else if (this._spy.calls.length === unexpectedCallCount) {
-      throw new FunctionCallCountMatchError(
-        this._spy,
-        false,
-        unexpectedCallCount,
-        SpyCallCountType.Exactly
-      );
-    }
+    this._throwCallCountError(
+      count => count === unexpectedCallCount,
+      unexpectedCallCount,
+      SpyCallCountType.Exactly,
+      false
+    );
 
     return new FunctionSpyCallCountMatcher();
   }
 
   public greaterThan(minimumCallCount: number): FunctionSpyCallCountMatcher {
-    if (minimumCallCount < 1) {
-      throw new TypeError("minimumCallCount must be greater than 0.");
-    }
+    this._validateCallCount(minimumCallCount, "minimumCallCount");
 
-    if (
-      this._expectedArguments &&
-      this._getArgumentsMatching(this._expectedArguments).length <=
-        minimumCallCount
-    ) {
-      throw new FunctionCallCountMatchError(
-        this._spy,
-        true,
-        minimumCallCount,
-        SpyCallCountType.GreaterThan,
-        this._expectedArguments
-      );
-    } else if (this._spy.calls.length <= minimumCallCount) {
-      throw new FunctionCallCountMatchError(
-        this._spy,
-        true,
-        minimumCallCount,
-        SpyCallCountType.GreaterThan
-      );
-    }
+    this._throwCallCountError(
+      count => count <= minimumCallCount,
+      minimumCallCount,
+      SpyCallCountType.GreaterThan,
+      true
+    );
 
     return new FunctionSpyCallCountMatcher();
   }
 
   public lessThan(maximumCallCount: number): FunctionSpyCallCountMatcher {
-    if (maximumCallCount < 1) {
-      throw new TypeError("maximumCallCount must be greater than 0.");
+    this._validateCallCount(maximumCallCount, "maximumCallCount");
+
+    this._throwCallCountError(
+      count => count >= maximumCallCount,
+      maximumCallCount,
+      SpyCallCountType.LessThan,
+      true
+    );
+
+    return new FunctionSpyCallCountMatcher();
+  }
+
+  private _validateCallCount(callCount: number, callCountName: string) {
```

Comment:
This can probably be named more generically, just a method that throws a `TypeError` if a number is less than 0 (or even better, less than `x`)